keanu-project/src/test/java/io/improbable/keanu/algorithms/mcmc/proposal/GaussianProposalDistributionTest.java

```diff
@@ -0,0 +1,67 @@
+package io.improbable.keanu.algorithms.mcmc.proposal;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.when;
+
+import org.apache.commons.math3.stat.correlation.Covariance;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import io.improbable.keanu.distributions.continuous.Gaussian;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.probabilistic.GaussianVertex;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GaussianProposalDistributionTest {
+
+    public Proposal proposal;
+    DoubleTensor currentState = DoubleTensor.create(4.2, 42.0).transpose();
+    DoubleTensor proposedState = DoubleTensor.create(4.3, 43.0).transpose();
+
+    @Mock
+    public GaussianVertex vertex1;
+    @Mock
+    public GaussianVertex vertex2;
+    private GaussianProposalDistribution proposalDistribution;
+    private DoubleTensor sigma;
+
+    @Before
+    public void setUpProposalDistribution() throws Exception {
+        sigma = DoubleTensor.scalar(1.);
+        proposalDistribution = new GaussianProposalDistribution(sigma);
+    }
+
+    @Before
+    public void setRandomSeed() throws Exception {
+        KeanuRandom.setDefaultRandomSeed(0);
+    }
+
+    @Before
+    public void setUpMocks() throws Exception {
+        when(vertex1.getValue()).thenReturn(DoubleTensor.scalar(currentState.getValue(0)));
+        when(vertex2.getValue()).thenReturn(DoubleTensor.scalar(currentState.getValue(1)));
+
+        proposal = new Proposal();
```

Comment:
Should the proposal setup be done in its own function (given the name of the function is to do with Mocks and this looks like just the test input).

Reply:
The order is important (I need to set up the mocks before the proposal) so they can't go in separate `@Before` methods.

I've renamed the method `setUpProposal` which is more accurate and hopefully addresses your point.